resetAndProvisionYubiKey.sh

```diff
@@ -69,10 +67,34 @@ ykman piv change-management-key \
 echo
 echo
 green_echo "STEP 4 - Generating & Setting A PUK And PIN"
-echo "  PUK:${puk}"
-echo "  PIN:${pin}"
+pin_pattern=^[a-zA-Z0-9]{6}$
+puk_pattern=^[a-zA-Z0-9]{8}$
+echo "  Use your personal vault to generate (and save) a PIN and a PUK."
+grey_echo "  The PUK must be 8 characters long and must only contain numbers, uppercase letters"
+grey_echo "  and lowercase letters."
+grey_echo "  The PIN must be 6 characters long and must only contain numbers, uppercase letters"
+grey_echo "  and lowercase letters."
+while true; do
+    echo
+    read -s -p "    Enter PUK: " puk
+    [[ $puk =~ $puk_pattern ]] && green_echo "OK" && break
+    echo
+    yellow_echo "    The PUK must be 8 characters long and only contain the following characters: a-zA-Z0-9"
+    echo "    Please try again"
+done
+while true; do
+    echo
+    read -s -p "    Enter PIN: " pin
+    [[ $pin =~ $pin_pattern ]] && green_echo "OK" && break
+    echo
+    yellow_echo "    The PIN must be 6 characters long and only contain the following characters: a-zA-Z0-9"
+    echo "    Please try again"
+done
+echo
+echo "  PUK: ${puk}"
+echo "  PIN: ${pin}"
```

Comment:
I feel like we do not need to display PIN and PUK with the new changes. What do you think?